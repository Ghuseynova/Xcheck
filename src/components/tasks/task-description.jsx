import React from 'react';
import 'antd/dist/antd.css';
import PropTypes from 'prop-types';
import { Descriptions } from 'antd';

const TaskDescription = (props) => {
  const task = props.task;
  const { title, author, taskScore, status, description } = task;

  return (
    <>
      <Descriptions column={1} bordered>
        <Descriptions.Item label="Task title">{title}</Descriptions.Item>
        <Descriptions.Item label="Author">{author}</Descriptions.Item>
        <Descriptions.Item label="Max score">{taskScore}</Descriptions.Item>
        <Descriptions.Item label="Status">{status}</Descriptions.Item>
        <Descriptions.Item label="Description">{description}</Descriptions.Item>
        <Descriptions.Item label="Link to this task "></Descriptions.Item>
      </Descriptions>
      <Descriptions title="User Info" column={1} bordered>
        <Descriptions.Item label="Task title">
          Task title Ефеф мидпфлм омфжлм тмжф
        </Descriptions.Item>
        <Descriptions.Item label="Author">ридм\ичвы осмывмо жлмижылвом</Descriptions.Item>
        <Descriptions.Item label="Max score">1000</Descriptions.Item>
      </Descriptions>
    </>
  );
};

TaskDescription.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  taskScore: PropTypes.number.isRequired,
  status: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

export default TaskDescription;
