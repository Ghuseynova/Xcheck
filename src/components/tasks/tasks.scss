.ant-descriptions-view .ant-descriptions-item-label {
  width: 20%;
}
