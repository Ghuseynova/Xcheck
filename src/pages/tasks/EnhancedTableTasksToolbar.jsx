import React from 'react';
import Typography from '@material-ui/core/Typography';
import Tooltip from '@material-ui/core/Tooltip';
import IconButton from '@material-ui/core/IconButton';
import Toolbar from '@material-ui/core/Toolbar';
import clsx from 'clsx';
import ArchiveIcon from '@material-ui/icons/Archive';
import { lighten, makeStyles } from '@material-ui/core/styles';
import AddIcon from '@material-ui/icons/Add';
import EditIcon from '@material-ui/icons/Edit';
import RemoveRedEyeIcon from '@material-ui/icons/RemoveRedEye';

const useToolbarStyles = makeStyles((theme) => ({
  root: {
    paddingLeft: theme.spacing(2),
    paddingRight: theme.spacing(1),
  },
  highlight:
    theme.palette.type === 'light'
      ? {
          color: theme.palette.secondary.main,
          backgroundColor: lighten(theme.palette.secondary.light, 0.9),
        }
      : {
          color: theme.palette.text.primary,
          backgroundColor: theme.palette.secondary.dark,
        },
  title: {
    flex: '1 1 100%',
  },
}));

const EnhancedTableToolbar = (props) => {
  const classes = useToolbarStyles();
  const { numSelected } = props;

  const renderSwitch = (param) => {
    switch (param) {
      case 0:
        return (
          <Tooltip title="Add">
            <IconButton aria-label="add item">
              <AddIcon />
            </IconButton>
          </Tooltip>
        );
      case 1:
        return (
          <>
            <Tooltip title="EditIcon">
              <IconButton aria-label="editIcon">
                <EditIcon />
              </IconButton>
            </Tooltip>
            <Tooltip title="Archive">
              <IconButton aria-label="archiveIcon">
                <ArchiveIcon />
              </IconButton>
            </Tooltip>
            <Tooltip title="Open">
              <IconButton aria-label="Open">
                <RemoveRedEyeIcon />
              </IconButton>
            </Tooltip>
          </>
        );
      default:
        return (
          <>
            <Tooltip title="Archive">
              <IconButton aria-label="archiveIcon">
                <ArchiveIcon />
              </IconButton>
            </Tooltip>
          </>
        );
    }
  };

  return (
    <Toolbar
      className={clsx(classes.root, {
        [classes.highlight]: numSelected > 0,
      })}
    >
      {numSelected > 0 > 2 ? (
        <Typography className={classes.title} color="inherit" variant="subtitle1" component="div">
          {numSelected} selected
        </Typography>
      ) : (
        <Typography className={classes.title} variant="h6" id="tableTitle" component="div">
          Tasks
        </Typography>
      )}

      {renderSwitch(numSelected)}
    </Toolbar>
  );
};

export default EnhancedTableToolbar;
