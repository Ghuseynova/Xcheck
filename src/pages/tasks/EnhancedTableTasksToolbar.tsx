import React from 'react';
import Typography from '@material-ui/core/Typography';
import Tooltip from '@material-ui/core/Tooltip';
import IconButton from '@material-ui/core/IconButton';
import Toolbar from '@material-ui/core/Toolbar';
import clsx from 'clsx';
import ArchiveIcon from '@material-ui/icons/Archive';
import AddIcon from '@material-ui/icons/Add';
import EditIcon from '@material-ui/icons/Edit';
import RemoveRedEyeIcon from '@material-ui/icons/RemoveRedEye';
import PropTypes from 'prop-types';
import { useToolbarStyles } from './styles';

const EnhancedTableToolbar = (props) => {
  const classes = useToolbarStyles();
  const { numSelected } = props;

  const renderSwitch = (param) => {
    switch (param) {
      case 0:
        return (
          <Tooltip title="Add">
            <IconButton aria-label="add item">
              <AddIcon />
            </IconButton>
          </Tooltip>
        );
      case 1:
        return (
          <>
            <Tooltip title="EditIcon">
              <IconButton aria-label="editIcon">
                <EditIcon />
              </IconButton>
            </Tooltip>
            <Tooltip title="Archive">
              <IconButton aria-label="archiveIcon">
                <ArchiveIcon />
              </IconButton>
            </Tooltip>
            <Tooltip title="Open">
              <IconButton aria-label="Open">
                <RemoveRedEyeIcon />
              </IconButton>
            </Tooltip>
          </>
        );
      default:
        return (
          <>
            <Tooltip title="Archive">
              <IconButton aria-label="archiveIcon">
                <ArchiveIcon />
              </IconButton>
            </Tooltip>
          </>
        );
    }
  };

  return (
    <Toolbar
      className={clsx(classes.root, {
        [classes.highlight]: numSelected > 0,
      })}
    >
      {numSelected > 0 ? (
        <Typography className={classes.title} color="inherit" variant="subtitle1" component="div">
          {numSelected} selected
        </Typography>
      ) : (
        <Typography className={classes.title} variant="h6" id="tableTitle" component="div">
          Tasks
        </Typography>
      )}

      {renderSwitch(numSelected)}
    </Toolbar>
  );
};

EnhancedTableToolbar.propTypes = {
  numSelected: PropTypes.number.isRequired,
};

export default EnhancedTableToolbar;
