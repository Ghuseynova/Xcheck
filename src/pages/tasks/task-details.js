import React from 'react';
import { PageHeader, Button } from 'antd';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import TaskDescription from '../../components/tasks/task-description';

export const TaskDetails = (props) => {
  const { tasks } = props;
  return (
    <div className="wrapper">
      <PageHeader className="site-page-header" title="Task details" />
      <Button type="primary">
        <Link to="/tasks/">Back</Link>
      </Button>
      {tasks?.formValues && tasks.formValues[0] && <TaskDescription task={tasks.formValues[0]} />}
    </div>
  );
};

TaskDetails.propTypes = {
  tasks: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

const mapStateToProps = ({ tasks }) => ({
  tasks,
});

export default connect(mapStateToProps)(TaskDetails);
