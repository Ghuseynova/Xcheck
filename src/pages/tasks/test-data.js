const task = {
	formValue: [{
		"taskScore": 100,
		"title": "Task 1",
		"state": "draft",
		"author": "viktorsipach",
		"description": "Task 1 description",
		"items": [
			{
				"id": "task-1-bp11",
				"title": "basic_p1",
				"category": "basic",
				"score": "10",
				"description": "basic_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-bp21",
				"title": "basic_p2",
				"category": "basic",
				"score": "20",
				"description": "basic_p2 - You need to do some things",
				"mentorCheck": false
			},
			{
				"id": "task-1-ep11",
				"title": "extra_p1",
				"category": "extra",
				"score": "30",
				"description": "extra_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-ep21",
				"title": "extra_p2",
				"category": "extra",
				"score": "50",
				"description": "extra_p2 - You need to do some things",
				"mentorCheck": false
			},
			{
				"id": "task-1-fp11",
				"title": "fines_p1",
				"category": "fines",
				"score": "10",
				"description": "fines_p1 - You need to do some things",
				"mentorCheck": true
			},
			{
				"id": "task-1-fp21",
				"title": "fines_p2",
				"category": "fines",
				"score": "10",
				"description": "fines_p2 - You need to do some things",
				"mentorCheck": false
			}
		],
		"taskId": "task-11",
		"id": "task-11"
	}]
};

export default task;