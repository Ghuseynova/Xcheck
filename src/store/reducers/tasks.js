import { POST_TASK_SESSIONS_SUCCESS, POST_TASK_SESSIONS_FAILURE } from '../actions/types';

const tasks = (state = [], action) => {
  switch (action.type) {
    case POST_TASK_SESSIONS_SUCCESS:
      return {
        ...state,
      };

    case POST_TASK_SESSIONS_FAILURE:
      return state;
    default:
      return state;
  }
};

export default tasks;
