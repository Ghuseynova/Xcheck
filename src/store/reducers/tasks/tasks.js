import {
  POST_TASK_SESSIONS_SUCCESS,
  POST_TASK_SESSIONS_FAILURE,
  GET_TASKSTABLE_SESSIONS_SUCCESS,
  GET_TASKSTABLE_SESSIONS_FAILURE,
  GET_TASK_BY_TITLE_SUCCESS,
  GET_TASK_BY_TITLE_FAILURE,
  UPDATE_TASK_SESSION_SUCCESS,
  UPDATE_TASK_SESSION_FAILURE,
} from '../../actions/types/task';

const initialState = {
  formValues: {},
  currentTask: {},
};

const tasks = (state = initialState, action) => {
  switch (action.type) {
    case GET_TASKSTABLE_SESSIONS_SUCCESS:
      return {
        ...state,
        formValues: action.payload,
      };

    case GET_TASKSTABLE_SESSIONS_FAILURE:
      return state;

    case GET_TASK_BY_TITLE_SUCCESS:
      return {
        ...state,
        currentTask: action.payload,
      };
    case GET_TASK_BY_TITLE_FAILURE:
      return {
        ...state,
      };

    case UPDATE_TASK_SESSION_SUCCESS:
      return {
        ...state,
        formValues: {},
      };

    case UPDATE_TASK_SESSION_FAILURE:
      return state;

    case POST_TASK_SESSIONS_SUCCESS:
      return {
        ...state,
      };

    case POST_TASK_SESSIONS_FAILURE:
      return state;

    default:
      return state;
  }
};

export default tasks;
