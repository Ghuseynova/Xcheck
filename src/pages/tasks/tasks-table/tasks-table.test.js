import React from 'react';
import { mount, shallow } from 'enzyme';
import { Router } from "react-router-dom";
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import configureStore from 'redux-mock-store';
import TasksTable from './tasks-table';
import tasks from './test-data';

describe('TasksTable page', () => {
	let store;
	const defaultProps = {
		getTasksTable: () => tasks[0],
		tasksTableData: tasks
	};
	beforeEach(() => {
		store = configureStore()({ tasksTableData: tasks });
	});

	it('toMatchSnapshot TasksTable', () => {
		const tree = shallow(<Router history={createMemoryHistory()}><TasksTable {...defaultProps} /></Router>);
		expect(tree).toMatchSnapshot();
	});

	it('Should render header, button', () => {
		const wrapper = mount(
			<Router history={createMemoryHistory()}>
				<Provider store={store}>
					<TasksTable {...defaultProps} />
				</Provider>
			</Router>
		)
		const header = wrapper.find('.ant-page-header-heading-title');
		const buttonAdd = wrapper.find('.ant-btn-primary');
		expect(header.length).toBe(1);
		expect(buttonAdd.length).toBe(1);
	});
});