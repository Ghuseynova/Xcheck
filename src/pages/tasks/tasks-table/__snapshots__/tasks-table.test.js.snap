// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksTable page toMatchSnapshot TasksTable 1`] = `ShallowWrapper {}`;
